src/legacy/ui/public/agg_types/controls/string.tsx

```diff
@@ -40,6 +40,11 @@ function StringParamEditor({
     [value]
   );
 
+  const onChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    const textValue = ev.target.value;
+    setValidity(aggParam.required ? !!textValue : true);
```

Comment:
I think that we shouldn't call `setValidity` here, for the same reason as I outlined in the other code review: https://github.com/elastic/kibana/pull/35478#discussion_r278935695